src/indexer.cc

```diff
@@ -830,6 +831,125 @@ class IndexDataConsumer : public index::IndexDataConsumer {
+              }
+              return RecursiveASTVisitor::VisitFieldDecl(decl);
+            }
+            bool VisitCXXConstructorDecl(CXXConstructorDecl* decl) {
+              for (const auto& init : decl->inits()) {
+                if (init->isMemberInitializer()) {
+                  if (auto* defaultInit = dyn_cast<CXXDefaultInitExpr>(init->getInit())) {
+                    AddDataFlow(init->getMember(), defaultInit->getExpr());
+                  } else {
+                    AddDataFlow(init->getMember(), init->getInit());
+                  }
+                }
+              }
+              return RecursiveASTVisitor::VisitCXXConstructorDecl(decl);
+            }
+            bool VisitCXXDefaultInitExpr(CXXDefaultInitExpr* expr) {
+              AddDataFlow(expr->getField(), expr->getExpr());
+              return RecursiveASTVisitor::VisitCXXDefaultInitExpr(expr);
+            }
+
+            bool VisitCallExpr(CallExpr* expr) {
+              if (FunctionDecl* decl = expr->getDirectCallee()) {
+                int len = std::min(expr->getNumArgs(), decl->getNumParams());
+                for (int i = 0; i < len; ++i) {
+                  AddDataFlow(decl->getParamDecl(i), expr->getArg(i));
+                }
+              }
+              return RecursiveASTVisitor::VisitCallExpr(expr);
+            }
+            bool VisitCXXConstructExpr(CXXConstructExpr* expr) {
+              if (CXXConstructorDecl* decl = expr->getConstructor()) {
+                int len = std::min(expr->getNumArgs(), decl->getNumParams());
+                for (int i = 0; i < len; ++i) {
+                  AddDataFlow(decl->getParamDecl(i), expr->getArg(i));
+                }
+              }
+              return RecursiveASTVisitor::VisitCXXConstructExpr(expr);
+            }
+
+            bool VisitReturnStmt(ReturnStmt* stmt) {
+              if (Expr* retVal = stmt->getRetValue()) {
+                if (const FunctionDecl* func = dyn_cast<FunctionDecl>(root))
+                  AddDataFlow(func, retVal);
+              }
+              return RecursiveASTVisitor::VisitReturnStmt(stmt);
+            }
+
+          private:
+            void AddDataFlow(const Expr* target, const Expr* source) {
+              if (const auto* targetDeclRef = dyn_cast<DeclRefExpr>(target->IgnoreParenCasts())) {
+                AddDataFlow(targetDeclRef->getDecl(), source);
+              } else if (const auto* targetFieldRef = dyn_cast<MemberExpr>(target->IgnoreParenCasts())) {
+                AddDataFlow(targetFieldRef->getMemberDecl(), source);
+              }
+            }
+            void AddDataFlow(const ValueDecl* target, const Expr* source) {
+              auto* var = &db->ToVar(parent->GetUsr(target));
+              var->data_flow_into.push_back(GetDataFlow(source));
+            }
+            void AddDataFlow(const FunctionDecl* target, const Expr* source) {
+              auto* func = &db->ToFunc(parent->GetUsr(target));
```

Comment:
Since we don't discourage uses of non-const references and `ToFunc` returns a non-const reference, we don't need to convert `func` to a pointer.